telemetry/telemetry.proto

```diff
@@ -4,6 +4,57 @@ import 'nanopb.proto';
 
 option java_outer_classname="TelemetryProto";
 
-message Telemetry {
-  optional int32 timestamp_ms = 1;
+//Should be sent only after device resets
+message NameIDMessage { //Send the device ID with its corresponding name
+	optional uint32 id = 1;
+	optional string name = 2 [(nanopb).max_size = 32];
+}
+
+message StatisticalAggregate {
+  optional uint32 numSamples = 1;
+  optional uint32 min = 2;
+  optional uint32 max = 3;
+  optional uint32 mean = 4;
+  optional uint32 stdev = 5;
+}
+
+message GeneralReading { //this might want a more specific name?
+	enum READING_TYPE {
+		VOLTAGE = 1;
+		CURRENT = 2;
+		TEMPERATURE = 3;
+	}
+	optional READING_TYPE type = 1;
```

Comment:
I think if you included that kind of data, it would be in the metadata. Measurement is well defined enough that stuff like resolution, limits, and accuracy might also make sense. However, Measurement doesn't have an idea of frequency, so that would need to be added. You should also separate frequency / period of incoming data (for example, CAN messages reporting some value) from outgoing data (how often Measurement messages are sent, this is less useful for debugging car systems, though may be useful for flagging lost packets / etc).

Whether to send metadata over the air is a trade-off between use of bandwidth and what your sources of information are. That's a design decision that also depends on how you want to structure your sources of information.